### PR TITLE
Add support for qframework from interface

### PR DESCRIPTION
**Pull request by jmazanec15**
### Description
Adds support for parsing qframework for faiss in the interface for IVF and HNSW. Does not yet integrate them with the framework.

### Check List
- [X] New functionality includes testing.
- [X] Commits are signed per the DCO using `--signoff`.

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/k-NN/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: src/main/java/org/opensearch/knn/index/engine/qframe/QuantizationConfigParser.java
Comment (heemin32) [inline src/main/java/org/opensearch/knn/index/engine/qframe/QuantizationConfigParser.java:17]: I think it would be better to have all these methods inside `QuantizationConfig` class itself?
```
// Creation from csv
QuantizationConfig config = QuantizationConfig.Builder.newInstance().addCsv(csv).build();
// or
QuantizationConfig config = new QuantizationConfig(csv);

// Back to csv
String configInString = config.toCsv();
```

Comment (jmazanec15) [reply src/main/java/org/opensearch/knn/index/engine/qframe/QuantizationConfigParser.java:17]: I decided to keep QuantizationConfig simple and have other class, as @shatejas had suggested in https://github.com/opensearch-project/k-NN/pull/1969#discussion_r1718900321 and in the field mapping parser changes. Im good with either, but dont see a major benefit for one vs other.

Comment (heemin32) [reply src/main/java/org/opensearch/knn/index/engine/qframe/QuantizationConfigParser.java:17]: This is more like we are having serialization and deserialization method inside KNNBuilderQuery and few other requests classes rather than having a separate static methods.

Those methods are tightly coupled with what field we have in the class. Whenever you make change in those field, you have to modify these methods. Having them in multiple classes is like broadening the scope of changes from one class to multiple classes whenever there is an update.

Comment (shatejas) [reply src/main/java/org/opensearch/knn/index/engine/qframe/QuantizationConfigParser.java:17]: In general serialization and deserialization can be expensive, this code doesn't seem to super expensive. Having said that, the question would be, what part of code base are we okay having access to this especially if we are not using java serialization or something with requires this to be in the POJO. Adding it in POJO itself, gives an easy access. on contrary client has to be mindful to use this parser and is not an easy access also keeping the responsibility different. 

These are my two cents, its trivial enough and I am okay if maintainers feel okay about adding it in pojo

Comment (heemin32) [reply src/main/java/org/opensearch/knn/index/engine/qframe/QuantizationConfigParser.java:17]: Fair enough. Will just leave it to the author of the code to decide.
However, I still believe it would be natural to have them inside Config class. Not this case right now but suppose if QuantizationConfig has child classes for example, then this approach might get complicated.

**Review by heemin32 — APPROVED** [review]

**Review by junqiu-lei — APPROVED** [review]

File: src/main/java/org/opensearch/knn/index/engine/qframe/QuantizationConfigParser.java
Comment (jmazanec15) [reply src/main/java/org/opensearch/knn/index/engine/qframe/QuantizationConfigParser.java:17]: Ill keep as is for now.

**Pull request merged**
